### PR TITLE
Add maximum_matching_labels input to enforce upper bound on labels

### PR DESCRIPTION
**Pull request by ludeeus**
## Summary
This PR adds a new optional `maximum_matching_labels` input to the action that allows users to enforce an upper bound on the number of matching labels. Combined with the existing "at least one" requirement, this enables use cases like requiring exactly one label from a set.

## Key Changes
- **New input parameter**: Added `maximum_matching_labels` to `action.yml` as an optional input that defaults to the number of supplied labels
- **Validation logic**: Implemented validation in `action.js` to ensure the input is a positive integer
- **Enforcement**: Added check to fail the action when matching labels exceed the specified maximum
- **Documentation**: Updated README with explanation of the new parameter and added example for "exactly one label" use case
- **Test coverage**: Added comprehensive test cases covering:
  - Default behavior (no cap enforcement)
  - Exceeding the maximum (failure case)
  - Boundary conditions (equal to maximum passes)
  - Invalid input validation (non-integer, zero, negative, decimal values)
  - Interaction with existing "no matching labels" error
- **CI workflow**: Added test job to verify the feature works in practice

## Implementation Details
- The maximum defaults to the count of required labels, making it a no-op unless explicitly set to a lower value
- Input validation uses regex to ensure only positive integers are accepted
- Error messages clearly indicate both the count found and the maximum allowed
- The maximum matching labels check runs after the minimum (at least one) check, so the "no matching labels" error takes precedence

https://claude.ai/code/session_0197be94KNgmsGv5RcrV2F1y

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,6 +116,28 @@ The example below requires the pull request to have at least one **type** label
...
 </details>
 
+### Requiring exactly one label
+
+Use [`maximum_matching_labels`](#maximum_matching_labels) to require **exactly one** of the listed labels — for example exactly one priority label.
+
+<details>
+<summary>More details and example</summary>
+
+The action already requires **at least one** of the listed labels. Setting `maximum_matching_labels: 1` adds the upper bound, so the step passes only when exactly one of the labels is present — no follow-up step or `continue-on-error` needed.
```
Comment (ludeeus) [inline README.md]:
```suggestion
The action already requires **at least one** of the listed labels. Setting `maximum_matching_labels: 1` adds the upper bound, so the step passes only when exactly one of the labels is present.
```
